### PR TITLE
Update block-filters.md

### PR DESCRIPTION
**Pull request by davy440**
## What?
Describing the `blocks.getBlockAttributes` filter

## Why?
The documentation needed updation regarding the filter. It could be helpful exploring further applications of the filter hook.

## How?
Currently, there is no documentation for the filter hook.

## Testing Instructions
1. Set up the block development environment using @wordpress/scripts package.
2. Setup an entry-point for webpack in the package.
3. Add the code to the entry pint.
4. Save and compile.

### CODE REVIEW

**Comment by akasunil** [comment]
Thank you for your contribution! 

Is it possible for us to utilize the same simple example function something like other filters example in the same readme? I think this one is a little too technical; since the purpose of this is just to show how to apply the filter, a simple function would do the job.

**Review by davy440 — COMMENTED** [review]
Modified the example and provided additional info

**Comment by davy440** [comment]
@akasunil Thanks for the feedback. I have modified the example and provided additional info regarding the filter.
Please review.

Thanks,
Divjot

File: docs/reference-guides/filters/block-filters.md [hunk 1]
```diff
@@ -294,6 +294,31 @@ Used to filter an individual transform result from block transformation. All of
...
+    if(blockType['name'] === 'core/paragraph') {
+        blockAttributes['lock'] = {move: true}
+    }
+    return blockAttributes;
+}
+
+// Call the filter
+wp.hooks.addFilter(
+    'blocks.getBlockAttributes',
+    'it-residence/lock-paragraphs',
```
Comment (akasunil) [inline docs/reference-guides/filters/block-filters.md]:
```suggestion
    'my-plugin/lock-paragraphs',
```

File: docs/reference-guides/filters/block-filters.md [hunk 2]
```diff
@@ -294,6 +294,31 @@ Used to filter an individual transform result from block transformation. All of
...
+```js
+// Our filter function
+function lockParagraphs( blockAttributes, blockType, innterHTML, attributes  ) {
+    if(blockType['name'] === 'core/paragraph') {
+        blockAttributes['lock'] = {move: true}
+    }
+    return blockAttributes;
+}
+
+// Call the filter
```
Comment (akasunil) [inline docs/reference-guides/filters/block-filters.md]:
```suggestion
// Add the filter
```

File: docs/reference-guides/filters/block-filters.md [hunk 3]
```diff
@@ -294,6 +294,31 @@ Used to filter an individual transform result from block transformation. All of
...
+- `blockAttributes` (`Object`): The attributes of the block node given its type.
+- `blockType` (`Object`): A block-type definition object.
+- `innerHTML` (`string`): Inner HTML content of the block.
+- `attributea` (`object`): Known attributes of the block.
+
+In the example below, we use the `blocks.getBlockAttributes` filter to lock the position of all paragraph blocks on a page.
+
+```js
+// Our filter function
+function lockParagraphs( blockAttributes, blockType, innterHTML, attributes  ) {
```
Comment (akasunil) [inline docs/reference-guides/filters/block-filters.md]:
```suggestion
function lockParagraphs( blockAttributes, blockType, innerHTML, attributes  ) {
```

File: docs/reference-guides/filters/block-filters.md [hunk 4]
```diff
@@ -294,6 +294,31 @@ Used to filter an individual transform result from block transformation. All of
 
 Called immediately after the default parsing of a block's attributes and before validation to allow a plugin to manipulate attribute values in time for validation and/or the initial values rendering of the block in the editor.
 
+The callback function for this filter accepts 4 parameters:
+- `blockAttributes` (`Object`): The attributes of the block node given its type.
+- `blockType` (`Object`): A block-type definition object.
+- `innerHTML` (`string`): Inner HTML content of the block.
+- `attributea` (`object`): Known attributes of the block.
```
Comment (akasunil) [inline docs/reference-guides/filters/block-filters.md]:
```suggestion
- `attributes` (`object`): Known attributes of the block.
```

File: docs/reference-guides/filters/block-filters.md [hunk 5]
```diff
@@ -294,6 +294,31 @@ Used to filter an individual transform result from block transformation. All of
...
+
+In the example below, we use the `blocks.getBlockAttributes` filter to lock the position of all paragraph blocks on a page.
+
+```js
+// Our filter function
+function lockParagraphs( blockAttributes, blockType, innterHTML, attributes  ) {
+    if(blockType['name'] === 'core/paragraph') {
```
Comment (akasunil) [inline docs/reference-guides/filters/block-filters.md]:
```suggestion
    if('core/paragraph' === blockType.name) {
```

**Comment by akasunil** [comment]
Thanks @davy440 for the ping. I have left couple of suggestions. Hope to get it addressed.

**Comment by davy440** [comment]
@akasunil Yes, I have provided a simpler example in my most recent commit. Is there something else I'm missing?

**Review by akasunil — CHANGES_REQUESTED** [review]

**Comment by davy440** [comment]
@akasunil thanks for the above fixes. Didn't pay attention to the above mentioned errors earlier.

File: docs/reference-guides/filters/block-filters.md [hunk 6]
```diff
@@ -294,6 +294,31 @@ Used to filter an individual transform result from block transformation. All of
 
 Called immediately after the default parsing of a block's attributes and before validation to allow a plugin to manipulate attribute values in time for validation and/or the initial values rendering of the block in the editor.
 
+The callback function for this filter accepts 4 parameters:
+- `blockAttributes` (`Object`): The attributes of the block node given its type.
+- `blockType` (`Object`): A block-type definition object.
+- `innerHTML` (`string`): Inner HTML content of the block.
+- `attributes` (`object`): Known attributes of the block.
```
Comment (akasunil) [inline docs/reference-guides/filters/block-filters.md]:
```suggestion
- `blockAttributes` (`Object`): All block attributes.
- `blockType` (`Object`): The block type.
- `innerHTML` (`string`): Raw block content.
- `attributes` (`object`): Known block attributes (from delimiters).
```

**Review by akasunil — CHANGES_REQUESTED** [review]
Thank you @davy440, These looks really good. just one final thing to address and then we are ready to go.

**Review by akasunil — APPROVED** [review]
LGTM!

**Comment by davy440** [comment]
@akasunil Is there any update regarding the PR? Please let me know if anything is required from my end.

Thank you,
Divjot

**Comment by akasunil** [comment]
Hi @davy440  Can you sync the PR with trunk?

**Comment by davy440** [comment]
@akasunil Just synced the fork with main branch. Showing 11 commits ahead of main branch. Let me know if anyhting else is required.

Thanks

**Pull request merged**